src/extension.ts

```diff
@@ -24,6 +25,56 @@ export function activate(context: vscode.ExtensionContext) {
 		(selectionOverride?: vscode.Selection) => continueEditing(outline, false, selectionOverride)
 	));
 
+	context.subscriptions.push(vscode.commands.registerCommand(COMMAND_MARKDOWN_COPILOT_EDITING_NAME_AND_SAVE, async () => {
+		const configuration = vscode.workspace.getConfiguration();
+		const textEditor = vscode.window.activeTextEditor;
+		if (textEditor === undefined) { return; }
+
+		function getDate(): string {
+			const date = new Date();
+			const year = date.getFullYear();
+			const month = String(date.getMonth() + 1).padStart(2, '0');
+			const day = String(date.getDate()).padStart(2, '0');
+			return `${year}-${month}-${day}`;
+		}
+
+		const stream = await createStream(
+			[
+				{ role: 'system', content: 'Provide a filename that best describes given content, using the same language as the content.' },
+				{
+					role: 'system', content: `Also follow the instruction: "${configuration.get<string>('markdown.copilot.instructions.titleMessage')}`
+				},
+				{ role: 'system', content: 'The filename must be concise, not contain any invalid filename characters, not contain any extensions, and not contain any whitespaces.' },
+				{ role: 'system', content: 'The filename must be returned in JSON format with the following format: {"filename":"{generated filename}"}' },
+				{ role: 'user', content: `Content:\n${textEditor.document.getText()}` }
+			] as OpenAIChatMessage[], {} as OpenAI.ChatCompletionCreateParamsStreaming);
+		let json = "";
+		for await (const chunk of stream) {
+			const chunkText = chunk.choices[0]?.delta?.content || '';
+			json += chunkText;
+		}
+		let filename = "";
+		try {
+			filename = JSON.parse(json).filename;
+		} catch {
+			vscode.window.showErrorMessage("Failed to generate a filename. Try again.");
```

Comment:
Not sure whether or not it is better that Copilot automatically tries again.

Reply:
It is expected that JSON parsing will fail rarely, so there is no need to retry 😉